python-modules/cis_profile/cis_profile/profile.py

```diff
@@ -100,6 +100,66 @@ def get_profile_from_file(self, user_structure_json_path):
             path = user_structure_json_path
         return DotDict(json.load(open(path)))
 
+    def merge(self, user_to_merge_in, publisher, level=None, _internal_level=None):
+        """
+        Recursively merge user attributes:
+        Merge a User object with another User object. This will override all fields from the current user by non-null
+        (non-None) fields from `user_to_merge_in` which have a publisher set to `publisher`.
+        Ex:
+        u_orig = User()
+        u_patch = User()
+        u_patch.timezone.value = "GMT -07:00 America/Los Angeles"
+
+        u_orig.merge(u_patch, publisher="hris")
+        print(u_orig.timezone.value)
+        # >>> GMT -07:00 America/Los Angeles
+
+        This is useful when updating existing users, rather than manually matching fields. Passing the publisher ensures
+        that we only patch the fields that are allowed for this publisher, as a safeguard, which ensures validation will
+        work
+
+        @user_to_merge_in User object the user to merge into the current user object
+        @publisher str the publisher name that attempts to merge data in. any data from `user_to_merge_in` that is not
+        originating from this publisher will be ignored
+        @level dict of an attribute. This can be user_to_merge_in.__dict__ for the top level (recurses through all
+        attributes). The level MUST be from user_to_merge_in, not from the original/current user.
+        @_internal_level str attribute name  of previous level attribute from the current user. Used internally.
+
+        This function always returns None and is always successful.
+        """
+        tomerge = []
+        # Default to top level
+        if level is None:
+            level = user_to_merge_in.__dict__
+
+        for attr in level.keys():
+            # If this is an internal class object, skip it
+            if attr.startswith("_") or not isinstance(level[attr], dict):
+                continue
+            # If we have no signature (or metadata in theory), this is not a "User attribute", keep doing deeper
+            if "signature" not in level[attr].keys():
+                self.merge(user_to_merge_in, publisher=publisher, level=level[attr], _internal_level=attr)
+            # Check if this is an attribute we want to merge back
+            elif level[attr]["signature"]["publisher"]["name"] == publisher:
+                tomerge.append(attr)
+
```

Comment:
```python
        for attr in tomerge:
            logger.debug("Merging in attribute {} for publisher {}".format(attr, publisher))

            try:
                # level[attr] is the patch/merged in current user
                # _intenal_level[attr] is in original user (self)
                # We skip null/None attributes even if the original/current user does not match (ie is not null)
                if level[attr].get("value") is not None or level[attr].get("values") is not None:
                     # copy everything including metadata and signatures
                    _intenal_level[attr] = level[attr]
```

Reply:
ok thats prettier. not in this case id drop the `try`